sky/serve/controller.py

```diff
@@ -64,17 +64,20 @@ def _run_autoscaler(self):
                     for info in replica_info
                 ]
                 logger.info(f'All replica info: {replica_info_dicts}')
-                scaling_option = self._autoscaler.evaluate_scaling(replica_info)
-                if (scaling_option.operator ==
-                        autoscalers.AutoscalerDecisionOperator.SCALE_UP):
-                    assert isinstance(scaling_option.target,
-                                      int), scaling_option
-                    self._replica_manager.scale_up(scaling_option.target)
-                elif (scaling_option.operator ==
-                      autoscalers.AutoscalerDecisionOperator.SCALE_DOWN):
-                    assert isinstance(scaling_option.target,
-                                      list), scaling_option
-                    self._replica_manager.scale_down(scaling_option.target)
+                scaling_options = self._autoscaler.evaluate_scaling(
+                    replica_info)
+                for scaling_option in scaling_options:
+                    logger.info(f'Scaling option received: {scaling_option}')
+                    if (scaling_option.operator ==
+                            autoscalers.AutoscalerDecisionOperator.SCALE_UP):
+                        assert (scaling_option.target is None or isinstance(
+                            scaling_option.target, dict)), scaling_option
+                        self._replica_manager.scale_up(scaling_option.target)
+                    elif (scaling_option.operator ==
+                          autoscalers.AutoscalerDecisionOperator.SCALE_DOWN):
+                        assert isinstance(scaling_option.target,
+                                          int), scaling_option
+                        self._replica_manager.scale_down(scaling_option.target)
```

Comment:
nit: `else: logger.error(...)`

Reply:
Shouldn't this be `except Exception as e:` to capture errors?

Reply:
I mean

```python
if op==up:
    # do sth
elif op==down:
    # do sth
else:
    raise ValueError(...) # or logger.error(...)